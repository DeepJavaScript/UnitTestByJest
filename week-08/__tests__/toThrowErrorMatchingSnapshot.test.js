class DisgustingFlavorError extends Error {}

function drinkFlavor(flavor) {
  if (flavor == "octopus") {
    throw new DisgustingFlavorError("yuck, octopus flavor");
  }
  console.log("drinkFlavor");
}

describe("toThrowErrorMatchingSnapshot", () => {
  it("throws on octopus", () => {
    function drinkOctopus() {
      drinkFlavor("octopus");
    }

    expect(drinkOctopus).toThrowErrorMatchingSnapshot();
  });

  it("throws on octopus", () => {
    function drinkOctopus() {
      drinkFlavor("octopus");
    }

    expect(drinkOctopus).toThrowErrorMatchingInlineSnapshot(
      `"yuck, octopus flavor"`
    );
  });
});
