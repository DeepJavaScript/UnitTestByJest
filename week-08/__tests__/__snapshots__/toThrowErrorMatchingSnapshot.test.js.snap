// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toThrowErrorMatchingSnapshot throws on octopus 1`] = `"yuck, octopus flavor"`;
